packages/eslint-plugin-office-addins/src/rules/no-navigational-load.ts

```diff
@@ -0,0 +1,88 @@
+import { TSESTree } from "@typescript-eslint/experimental-utils";
+import {
+  Reference,
+  Scope,
+  Variable,
+} from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+import { isLoadFunction, getPropertyNameInLoad } from "../utils/utils";
+import { isGetFunction } from "../utils/getFunction";
+import { getPropertyType, PropertyType } from "../utils/propertiesType";
+
+export = {
+  name: "no-navigational-load",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"problem",
+    messages: {
+      navigationalLoad:
+        "Calling load on the navigation property '{{loadValue}}' slows down your add-in.",
+    },
+    docs: {
+      description:
+        "Calling load on a navigation property causes unneeded data to load and slows down your add-in.",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Best Practices",
+      recommended: <false | "error" | "warn">false,
+      url: "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model#scalar-and-navigation-properties",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
+    function trimPropertyName(propertyName: string): string {
+      return propertyName.split("/").pop() ?? "";
+    }
+
+    function isNotScalarLoad(
```

Comment:
I prefer using a positive in method names and using a negation outside.  !isScalarLoad(...) is easier to read than isNotScalarLoad().

Also . . . what is a "ScalarLoad" seems like you're trying to combine different idea together that don't need to be.  I think it's clearer to test for isLoad and isScalar separately and then let a conditional check for both.